Source/CarthageKit/Project.swift

```diff
@@ -314,17 +304,53 @@ public final class Project { // swiftlint:disable:this type_body_length
 			}
 	}
 
+	/// Produces the sub dependencies of the given dependency. Uses the checked out directory if able
+	private func dependencySet(for dependency: Dependency, version: PinnedVersion) -> SignalProducer<Set<Dependency>, CarthageError> {
+		return self.dependencies(for: dependency, version: version, tryCheckoutDirectory: true)
+			.map { $0.0 }
+			.collect()
+			.map { Set($0) }
+			.concat(value: Set())
+			.take(first: 1)
+	}
+
 	/// Loads the dependencies for the given dependency, at the given version.
 	private func dependencies(for dependency: Dependency, version: PinnedVersion) -> SignalProducer<(Dependency, VersionSpecifier), CarthageError> {
+		return self.dependencies(for: dependency, version: version, tryCheckoutDirectory: false)
+	}
+
+	/// Loads the dependencies for the given dependency, at the given version. Optionally can attempt to read from the Checkout directory
+	private func dependencies(for dependency: Dependency, version: PinnedVersion, tryCheckoutDirectory: Bool) -> SignalProducer<(Dependency, VersionSpecifier), CarthageError> {
 		switch dependency {
 		case .git, .gitHub:
 			let revision = version.commitish
-			return self.cloneOrFetchDependency(dependency, commitish: revision)
+			let cartfileFetch: SignalProducer<Cartfile, CarthageError> = self.cloneOrFetchDependency(dependency, commitish: revision)
 				.flatMap(.concat) { repositoryURL in
 					return contentsOfFileInRepository(repositoryURL, Constants.Project.cartfilePath, revision: revision)
 				}
 				.flatMapError { _ in .empty }
 				.attemptMap(Cartfile.from(string:))
+
+			let cartfileSource: SignalProducer<Cartfile, CarthageError>
+			if tryCheckoutDirectory {
+				let dependencyURL = self.directoryURL.appendingPathComponent(dependency.relativePath)
+				cartfileSource = SignalProducer<Bool, NoError> {
+					var isDirectory: ObjCBool = false
+					return FileManager.default.fileExists(atPath: dependencyURL.path, isDirectory: &isDirectory) && isDirectory.boolValue
+				}
+				.flatMap(.concat) { directoryExists -> SignalProducer<Cartfile, CarthageError> in
+					if directoryExists {
+						return SignalProducer(result: Cartfile.from(file: dependencyURL.appendingPathComponent(Constants.Project.cartfilePath)))
+							.flatMapError { _ in .empty }
+					} else {
+						return cartfileFetch
```

Comment:
This covers the case where the checkout isn't there, like when we have a binary release. 

If we resolved #1466 (and solved the blocking issue that we build everything in the folder without verifying there were binary releases), then we wouldn't need this backup behavior anymore.